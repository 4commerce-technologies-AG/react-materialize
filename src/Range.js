import React from 'react';
import PropTypes from 'prop-types';

class Range extends React.Component {
  componentDidMount() {
    if (typeof M !== 'undefined') {
      this.instance = M.Range.init(this._range);
    }
  }

  componentWillUnmount() {
    this.instance && this.instance.destroy();
  }

  render() {
    return (
      <p className="range-field">
        <input
          type="range"
          ref={el => {
            this._range = el;
          }}
          {...this.props}
        />
      </p>
    );
  }
}

Range.propTypes = {
  min: PropTypes.string.isRequired,
  max: PropTypes.string.isRequired
};

export default Range;
