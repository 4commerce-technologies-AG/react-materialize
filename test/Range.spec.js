import React from 'react';
import { shallow } from 'enzyme';
import Range from '../src/Range';

describe('<Range />', () => {
  let wrapper;

  test('renders', () => {
    wrapper = shallow(<Range min="10" max="99" />);
    expect(wrapper).toMatchSnapshot();
  });
});
