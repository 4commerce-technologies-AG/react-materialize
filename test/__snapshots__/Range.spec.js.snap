// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Range /> renders 1`] = `
<p
  className="range-field"
>
  <input
    max="99"
    min="10"
    type="range"
  />
</p>
`;
